### ISSUE TITLE
catkin build does not follow symlinks properly

### ISSUE THREAD

**Issue by hgaiser** [catkin/catkin_tools#182]
I am working on OSX with ROS, but some packages can't compile there due to drivers that aren't available. For that reason, I am also using ubuntu in a virtual machine. To reduce duplication, I symbolically linked the `src` folder from OSX to the ubuntu guest machine. In that way, source code is the same in OSX and in ubuntu but both OS's have their own `build` and `devel` folder. Running `catkin build` in ubuntu somewhere inside the symlinked folder will try to erroneously use the build/devel from OSX, which is incompatible with the ubuntu version and will fail.

Long story short: running `catkin build` in `/home/hgaiser/catkin_ws/src<symlinked>/some_package` (where src is a symlink to `/media/psf/Home/catkin_ws/src`) is equal to `cd /media/psf/Home/catkin_ws && catkin build`, which doesn't work.

**Comment by wjwwood** [comment]
Is this a duplicate of https://github.com/catkin/catkin_tools/issues/91?

**Comment by jbohren** [comment]
@wjwwood I don't think this is an exact duplicate of #91 but it is related.

@hgaiser For now if you want to build from within your symlinked source tree after already sourcing your environment, you can run:

```
catkin build -w  ${CMAKE_PREFIX_PATH%%:*}
```

**Comment by jbohren** [comment]
@hgaiser Take a look at #186

**Comment by rukie** [comment]
Can someone elaborate on how to resolve this? I also am running a guest Ubuntu Virtual Machine that symbolically links to a host windows machine. 
Running the following command doesn't seem to make a difference.
```
catkin build -w ${CMAKE_PREFIX_PATH%%:*}
```
